File: src/codility/java/slice/MaxSliceSum.java
package codility.java.slice;

public class MaxSliceSum {

    public int solution(int[] A) {
        int maxSum = A[0];
        int currentSum = A[0];

        for (int i = 1;i< A.length;i++) {
            currentSum += A[i];
            currentSum = Math.max(A[i], currentSum);
            maxSum = Math.max(currentSum, maxSum);
        }
        return maxSum;
    }
}

File: src/codility/kotlin/MaxSliceSum.kt
package codility.kotlin

class MaxSliceSum {
    fun solution(A: IntArray): Int {
        var currentSum = A[0]
        var maxSum = A[0]

        for (i in 1 until A.size) {
            if (currentSum + A[i] > A[i]) {
                currentSum += A[i]
            } else {
                currentSum = A[i]
            }
            maxSum = Math.max(currentSum, maxSum)
        }
        return maxSum
    }

}